programs/bpf_loader/src/syscalls.rs

```diff
@@ -857,6 +854,14 @@ impl<'a> SyscallObject<BpfError> for SyscallCreateProgramAddress<'a> {
         memory_mapping: &MemoryMapping,
         result: &mut Result<u64, EbpfError<BpfError>>,
     ) {
+        let invoke_context = match self.invoke_context.try_borrow() {
```

Comment:
How about:
```suggestion
    let invoke_context = invoke_context
        .try_borrow()
        .map_err(|_| SyscallError::InvokeContextBorrowFailed)?;
```

Reply:
This function doesn't return anything, the result goes into the `result` argument.

Reply:
The other option is:
```
 let invoke_context = question_mark!(invoke_context
        .try_borrow()
        .map_err(|_| SyscallError::InvokeContextBorrowFailed), result);
```

Reply:
That might be better.

Reply:
Fixed now

Reply:
Yup, that is what I meant, thanks for the fix :-)